First docs for Microsoft identity web (#50)
MicrosoftDocs/microsoft-authentication-library-dotnet: microsoft-identity-web-articles/getting-started/scenarios.md

---

> title: MSAL.NET scenarios

Reviewer: ```suggestion
title: Microsoft Identity Web scenarios
```

> description: Learn the application scenarios and authentication flows supported by MSAL.NET

Reviewer: ```suggestion
description: Learn the application scenarios and authentication flows supported by Microsoft Identity Web.
```

services: active-directory
author: Dickson-Mwendia
manager: CelesteDG

ms.service: active-directory
ms.subservice: develop
ms.topic: reference
ms.workload: identity
ms.date: 03/17/2023
ms.author: dmwendia
ms.reviewer: localden
ms.custom: devx-track-csharp, aaddev, engagement-fy23

> # Customer intent: As an application developer, I want to know the application scenarios and authentication flows supported by MSAL.NET. 

Reviewer: ```suggestion
# Customer intent: As an application developer, I want to know the application scenarios and authentication flows supported by Microsoft Identity Web. 
```

---

# Microsoft identity web scenarios

## Introduction

> Microsoft.Identity.Web proposes a higher level API to protect web APIs, and acquire tokens in web apps, web APIs and services/daemon apps. You can choose to describe your application through a configuration file, or programmatically, or use a combination of both approaches.

Reviewer: ```suggestion
Microsoft.Identity.Web proposes a higher level API, over MSAL.NET, to protect web APIs, and acquire tokens in web apps, web APIs and services/daemon apps. You can choose to describe your application through a configuration file, or programmatically, or use a combination of both approaches.
```

## The Scenarios

Microsoft identity web is focused on services, and confidential client applications.

### Web app that signs in users and calls a web API on behalf of the user

To protect a web app (signing in the user) you'll use ASP.NET or ASP.NET Core with the ASP.NET Open ID Connect middleware. You'll reference the [Microsoft.Identity.Web](https://www.nuget.org/packages/Microsoft.Identity.Web) NuGet package if you use ASP.NET Core, and [Microsoft.Identity.Web.OWIN](https://www.nuget.org/packages/Microsoft.Identity.Web.OWIN) if you are still using ASP.NET (OWIN).

> If moreover, your web apps calls web APIs in the name of the user (or in its own name), you'll add the following NuGet packages:

Reviewer: ```suggestion
If moreover, your web apps call web APIs in the name of the user (or as itself), you'll add the following NuGet packages:
```
